### DISCUSSION TITLE
feature: ability to set variable with player param

### DISCUSSION THREAD

**Discussion by Spectreye** [AscensionGameDev/Intersect-Engine#2212] [Ideas]
**Is your feature request related to a problem? Please describe.**
It would be very nice if we could set variable with player param, there's a lot of possibilities behind.

here some exemple : Make Repeatable quest depending of the actual player level, or make a Potion wich restore Hp depending of the player level, you can even make spell  wich scale of multiple stats !

**Describe the solution you'd like**
The possibilities to set a variable with player param.

**Additional context**
I've made a screenshot of what it could looks like : 
![Player Param Value](https://user-images.githubusercontent.com/68560931/97471104-a9150980-1948-11eb-8d55-995485108851.png)

**Comment by WeylonSantana** [comment]
Bringing a complement from @Xyseer764 from #851

> * The Count of a specific Item
>   `This is also implemented in a non-modular way, where we just have access to this value in conditions. When for example I wanted to create a system to take half of a player's item, I found out that it's impossible without editing the source code or doing some extremely unscalable and slow implementation that takes the player's items one by one.`
> * Player X and Y position (global position would be better than local, creating a tickbox for choosing between global and local positions would be best)
>   `This allows the users to create zones or areas in which some event takes place.`
> * Free inventory slots
>   `Non-modular implementation in conditions currently.`
> 
> **Additional context** This feature requires the creation of a single drop-down menu and the hooking up of each of the listed variables to the options. Its dev cost is low, but the things it enables are general. Pretty much any developer would be happy to see this feature.

**Comment by WeylonSantana** [comment]
Bringing a complement from @AlexVild from #1110
>Event X
>Event Y

![image](https://user-images.githubusercontent.com/63019821/188622654-40e2d02b-7906-43a2-8c6f-1e5b0e35b25b.png)
![image](https://user-images.githubusercontent.com/63019821/188622604-a234f8dc-2387-43a3-8882-2f95c9acdb9b.png)
